arbitrator/src/arbitrator.cpp

```diff
@@ -194,4 +197,19 @@ namespace arbitrator
     {
         vehicle_state_.velocity = msg->twist.linear.x;
     }
+
+    void Arbitrator::lookupFrontBumperTransform() 
+    {
+        tf2_listener_.reset(new tf2_ros::TransformListener(tf2_buffer_));
+        tf2_buffer_.setUsingDedicatedThread(true);
```

Comment:
Does this create a whole new thread for doing the transform listening? Given that this function is called every pose_cb, it seems a bit excessive to re-initialize the listener every time, especially if it spins up a new thread. I've not worked too much with the tf2 server clients, so if this is standard practice I may be off base. But I think it could also make sense to just look up the transform once and cache it, as the transform itself should be a constant value.

Reply:
Since this transform represents the location of front bumper in the map, it needs to be updated with localization. I tried looking it up once, but it was not able to get the current location of the vehicle.

Reply:
Circling back on this, maybe I'm misunderstanding the whole transform tree system, but shouldn't it just be a relative value based on the urdf file? As in, the transform from rear-center axle to front bumper should be some fixed offset based on the vehicle length, and given a rear axle pose we only need to apply that fixed offset transform  (which could be looked up once and cached) to get the pose of the front bumper?

Am I misunderstanding the way the transforms are applied or stored?